src/WebJobs.Script.Grpc/MessageExtensions/GrpcMessageConversionExtensions.cs

```diff
@@ -222,6 +221,35 @@ internal static async Task<TypedData> ToRpcHttp(this HttpRequest request, ILogge
             return typedData;
         }
 
+        private static void PopulateHttpRouteDataAsParams(HttpRequest request, RpcHttp http, bool shouldUseNullableValueDictionary)
+        {
+            if (!request.HttpContext.Items.TryGetValue(HttpExtensionConstants.AzureWebJobsHttpRouteDataKey, out var routeData))
+            {
+                return;
+            }
+
+            if (routeData == null)
+            {
+                return;
+            }
+
+            var parameters = (Dictionary<string, object>)routeData;
```

Comment:
Can we change this to perform a safe cast and log if that fails instead? You can combine this into the condition above and that would eliminate the need for the `null` check, simplifying this code.

Reply:
Done.